src/index.js

```diff
@@ -1,40 +1,40 @@
 'use strict'
 
-const contentType = require('content-type')
-const getRawBody = require('raw-body')
+const rawBody = (req, { limit = 1048576 } = {}) =>
+  new Promise((resolve, reject) => {
+    const chunks = []
+    let bytes = 0
+    req
+      .on('error', reject)
+      .on('end', () => resolve(Buffer.concat(chunks, bytes)))
+      .on('data', chunk => {
+        chunks.push(chunk)
+        bytes += chunk.length
+        if (bytes > limit) {
+          reject(
+            new TypeError(`body size (${bytes}) is over the limit (${limit})`)
+          )
+        }
+      })
+  })
 
 const rawBodyMap = new WeakMap()
 
-const LIMIT = '1mb'
-
-const text = async (req, { limit = LIMIT, encoding } = {}) => {
-  const type = req.headers['content-type'] || 'text/plain'
-  const length = req.headers['content-length']
-
-  const body = rawBodyMap.get(req)
-  if (body) return body
-
-  if (encoding === undefined) {
-    encoding = contentType.parse(type).parameters.charset
-  }
-
-  const buffer = await getRawBody(req, { limit, length, encoding })
-  rawBodyMap.set(req, buffer)
-  return buffer.toString()
+const buffer = async (req, opts) => {
+  let body = rawBodyMap.get(req)
+  if (body !== undefined) return body
+  body = await rawBody(req, opts)
+  rawBodyMap.set(req, body)
+  return body
 }
 
-const buffer = async (req, { limit = LIMIT } = {}) => {
-  const length = req.headers['content-length']
-
-  const body = rawBodyMap.get(req)
-  if (body) return body
-
-  const buffer = await getRawBody(req, { limit, length })
-  rawBodyMap.set(req, buffer)
-
-  return buffer
-}
+const text = (req, opts) => buffer(req, opts).then(buffer => buffer.toString())
```

Comment:
An example for a JSDoc:

```suggestion
/**
 * Read the whole body as a string value
 *
 * @param {import('http').IncomingMessage} req the request
 * @param {Partial<Options>} [opts] parsing options
 */
const text = (req, opts) => buffer(req, opts).then(buffer => buffer.toString())
```

You can add examples, etc. I love good comments.

Reply:
By the way, one would argue that all your functions can take `Readable`. The fact it's an IncomingMessage is good but you don't really do anything with the IncomingMessage other than reading it as a stream. This could show that people can use it with any Readable, and simplify unit tests for this library, if necessary <picture data-single-emoji=":idea:" title=":idea:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..8DAAHBzqhxMfb5hx.DlDOTDwQN-9fxgcx-x9OZcRuUXh8rVaL4S9PLDjGEGxXSOnU3pFj2I-gpjioD2PyGPdAtN01XiHE79_iPCM-dhJlsxAaVzVu2bjuPCEsNYOo5Vr9BA.WrqBBIN2Y_wk9HpVnImS2g" alt=":idea:" align="absmiddle"></picture>

Reply:
That would look like:

```suggestion
/**
 * Read the whole body as a string value
 *
 * @param {import('stream').Readable} req the request (or any other stream)
 * @param {Partial<Options>} [opts] parsing options
 */
const text = (req, opts) => buffer(req, opts).then(buffer => buffer.toString())
```